plugins/modules/web_infrastructure/htpasswd.py

```diff
@@ -195,11 +200,11 @@ def check_file_attrs(module, changed, message):
 
 def main():
     arg_spec = dict(
-        path=dict(required=True, aliases=["dest", "destfile"]),
-        name=dict(required=True, aliases=["username"]),
-        password=dict(required=False, default=None, no_log=True),
-        crypt_scheme=dict(required=False, default="apr_md5_crypt"),
-        state=dict(required=False, default="present"),
+        path=dict(type='path', required=True, aliases=["dest", "destfile"]),
+        name=dict(type='str', required=True, aliases=["username"]),
+        password=dict(type='str', required=False, default=None, no_log=True),
+        crypt_scheme=dict(type='str', required=False, default="apr_md5_crypt", choices=["apr_md5_crypt", "des_crypt", "ldap_sha1", "plaintext"]),
```

Comment:
The code allows for values outside this list: https://github.com/ansible-collections/community.general/blob/5a427675da685250630b417874aa28b611c19700/plugins/modules/web_infrastructure/htpasswd.py#L126-L129

So adding the `choices` here is probably breaking backwards compatibility.

Reply:
oh, I missed that, thanks for spotting it.